packages/component_code_gen/README.md

```diff
@@ -0,0 +1,36 @@
+# AI Code Gen for Pipedream Components
+
+Generate components using OpenAI GPT-4.
+
+
+### Run
+
+```
+SCRIPT=generate_action.py
+APP=slack
+PROMPT="how to send myself a direct message?"
```

Comment:
`PROMPT` is actually a reserved variable in macOS and maybe other *nixes, so I would change this. I tried to set `PROMPT` but then still got:
```
❯ echo $PROMPT
${$((_p9k_on_expand()))+}%{${_p9k__raw_msg-}${_p9k__raw_msg::=}%}${(e)_p9k_t[7]}${_p9k__1-${${:-${_p9k__d::=0}${_p9k__rprompt::=${_p9k__1r-${${:-${_p9k__bg::=NONE}${_p9k__i::=0}${_p9k__sss::=${(Q)${:-"%\{%\}"}}}}+}}}${_p9k__e::=${${_p9k__1r:+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}0}}}+}${${_p9k__e:#00}:+${_p9k_t[$_p9k__n]/<_p9k__w>/$_p9k__w}${_p9k__c}%b%k%f${${:-${_p9k__w::=%b%k%f}${_p9k__sss::=%b%k%f}${_p9k__i::=1}${_p9k__bg:
}}${_p9k__e::=${${_p9k__1rcommand_execution_time+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}0}}}+}${${_p9k__e:#00}:+${_p9k_t[$_p9k__n]/<_p9k__w>/$_p9k__w}%ftook %b%k%F{003\}${_p9k__c}%b%k%F{003\}${${:-${_p9k__w::=%b%k%F{003\}}${_p9k__sss::=%b%k%F{003\}}${_p9k__i::=2}${_p9k__bg::=}}+}}${${:-"${${(%):-%j}:#0}"}:+${${:-${P9K_CONTENT::=""}${_p9k__n::=}${${${_p9k__bg:-0}:#NONE}:-${_p9k__n::=12}}${_p9k__n:=${${(M)${:-x$_p9k__bg}:#x(|0)}}}${_p9k__e::=${${_p9k__1rbackground_jobs+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}1}}}+}${${_p9k__e:#00}:+${_p9k_t[$_p9k__n]/<_p9k__w>/$_p9k__w}${_p9k__v}${${(M)_p9k__e:#11}:+ }${_p9k__c}%b%k%F{001\}${${:-${_p9k__w::=%b%k%F{001\}}${_p9k__sss::=%b%k%F{001\}}${_p9k__i::=3}${_p9k__bg::=}}+}}}${${:-${P9K_CONTENT::=14.19.0}${_p9k__n::=}${${${_p9k__bg:-0}:#NONE}:-${_p9k__n::=88}}${_p9k__n:=${${(M)${:-x$_p9k__bg}:#x(|0)}:+90}}${_p9k__}}${_p9k__e::=${${_p9k__1rasdf+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}1}}}+}${${_p9k__e:#00}:+${_p9k_t[$_p9k__n]/<_p9k__w>/$_p9k__w}${_p9k__v}${${(M)_p9k__e:#11}:+ }${_p9k__c}%b%k%F{002\}${${:-${_p9k__w::=%b%k%F{002\}}${_p9k__sss::=%b%k%F{002\}}${_p9k__i::=5}${_p9k__bg::=}}+}}${${:-${P9K_CONTENT::=7.13.4}${_p9k__n::=}${${${_p9k__bg:-0}:#NONE}:-${_p9k__n::=92}}${_p9k__n:=${${(M)${:-x$_p9k__bg}:#x(|0)}:+94}}${_p9k__n:=95}${_p9k_}}${_p9k__e::=${${_p9k__1rasdf+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}1}}}+}${${_p9k__e:#00}:+${_p9k_t[$_p9k__n]/<_p9k__w>/$_p9k__w}${_p9k__v}${${(M)_p9k__e:#11}:+ }${_p9k__c}%b%k%F{006\}${${:-${_p9k__w::=%b%k%F{006\}}${_p9k__sss::=%b%k%F{006\}}${_p9k__i::=5}${_p9k__bg::=}}+}}${${:-${P9K_CONTENT::=gke_development-235719_us-central1-a_development/dylburger}${_p9k__n::=}${${${_p9k__bg:-0}:#NONE}:-${_p9k__n::=16}}${_p9k__n:=${${(M)${:-x$_p9k__bg}:#x(|0)}:+18}}${_p9k__n:=19}${P9K_VISUAL_IDENTIFIER::=⎈}${_p9k__v::="○"}${_p9k__c::="${P9K_KUBECONTEXT_CLOUD_CLUSTER:-${P9K_KUBECONTEXT_NAME}}${${:-/$P9K_KUBECONTEXT_NAMESPACE}:#/default}"}${_p9k__}}${_p9k__e::=${${_p9k__1rkubecontext+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}1}}}+}${${_p9k__e:#00}:+${_p9k_t[$_p9k__n]/<_p9k__w>/$_p9k__w}%fat %b%k%F{005\}${_p9k__v}${${(M)_p9k__e:#11}:+ }${_p9k__c}%b%k%F{005\}${${:-${_p9k__w::=%b%k%F{005\}}${_p9k__sss::=%b%k%F{005\}}${_p9k__i::=23}${_p9k__bg::=}}+}}${${:-"${${(M)${#P9K_GCLOUD_PROJECT_NAME}:#0}:+$P9K_GCLOUD_ACCOUNT$P9K_GCLOUD_PROJECT_ID}"}:+${${:-${P9K_CONTENT::="${P9K_GCLOUD_ACCOUNT//\%/%%}:${P9K_GCLOUD_PROJECT_ID//\%/%%}"}${_p9k__n::=}${${${_p9k__bg:-0}:#NONE}:-${_p9k__n::=20}}${_p9k__n:=${${(M)${:-x$_p9k__bg}:#x(|0)}:+22}}${_p9k__n:=23}${P9K_VISUAL_IDENTIFIER::=G}${_p9k__v::=G}${_p9}}${_p9k__e::=${${_p9k__1rgcloud+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}1}}}+}${${_p9k__e:#00}:+${_p9k_t[$_p9k__n]/<_p9k__w>/$_p9k__w}${_p9k__v}${${(M)_p9k__e:#11}:+ }${_p9k__c}%b%k%F{004\}${${:-${_p9k__w::=%b%k%F{004\}}${_p9k__sss::=%b%k%F{004\}}${_p9k__i::=28}${_p9k__bg::=}}+}}}${${:-"$P9K_GCLOUD_PROJECT_NAME"}:+${${:-${P9K_CONTENT::="${P9K_GCLOUD_ACCOUNT//\%/%%}:${P9K_GCLOUD_PROJECT_ID//\%/%%}"}${_p9k__n::=}${${${_p9k__bg:-}}${_p9k__e::=${${_p9k__1rgcloud+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}1}}}+}${${_p9k__e:#00}:+${_p9k_t[$_p9k__n]/<_p9k__w>/$_p9k__w}${_p9k__v}${${(M)_p9k__e:#11}:+ }${_p9k__c}%b%k%F{004\}${${:-${_p9k__w::=%b%k%F{004\}}${_p9k__sss::=%b%k%F{004\}}${_p9k__i::=28}${_p9k__bg::=}}+}}}${${:-"${${(%):-%#}:#\#}"}:+${${:-${P9K_CONTENT::=%n@%m}${_p9k__n::=}${${${_p9k__bg:-0}:#NONE}:-${_p9k__n::=28}}${_p9k__n:=${${(M)${:-x$_p9k__bg}:#x(}}${_p9k__e::=${${_p9k__1rcontext+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}0}}}+}${${_p9k__e:#00}:+${_p9k_t[$_p9k__n]/<_p9k__w>/$_p9k__w}%fwith %b%k%F{007\}${_p9k__c}%b%k%F{007\}${${:-${_p9k__w::=%b%k%F{007\}}${_p9k__sss::=%b%k%F{007\}}${_p9k__i::=31}${_p9k__bg::=}}+}}}${${:-"${${(%):-%#}:#\%}"}:+${${:-${P9K_CONTENT::=%B%n@%m}${_p9k__n::=}${${${_p9k__bg:-0}:#NONE}:-${_p9k__n::=32}}${_p9k__n:=${${(M)${:-x$_p9k__bg}:#x(|0)}:+34}}$}}${_p9k__e::=${${_p9k__1rcontext+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}0}}}+}${${_p9k__e:#00}:+${_p9k_t[$_p9k__n]/<_p9k__w>/$_p9k__w}%fwith %b%k%F{001\}${_p9k__c}%b%k%F{001\}${${:-${_p9k__w::=%b%k%F{001\}}${_p9k__sss::=%b%k%F{001\}}${_p9k__i::=31}${_p9k__bg::=}}+}}}${${:-${P9K_CONTENT::=18:40:04}${_p9k__n::=}${${${_p9k__bg:-0}:#NONE}:-${_p9k__n::=36}}${_p9k__n:=${${(M)${:-x$_p9k__bg}:#x(|0)}:+38}}${_p9k__n:=39}${_p9k__c::="$}}${_p9k__e::=${${_p9k__1rtime+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}0}}}+}${${_p9k__e:#00}:+${_p9k_t[$_p9k__n]/<_p9k__w>/$_p9k__w}%fat %b%k%F{006\}${_p9k__c}%b%k%F{006\}${${:-${_p9k__w::=%b%k%F{006\}}${_p9k__sss::=%b%k%F{006\}}${_p9k__i::=42}${_p9k__bg::=}}+}}$_p9k__sss%b%k%f}}${_p9k__lprompt::=${_p9k__1l_frame-"%2F╭─"}${_p9k__1l-${${:-${_p9k__bg::=NONE}${_p9k__i::=0}${_p9k__sss::=%f }}+}${${:-${P9K_CONTENT::="%{d%}${:-"%F{004}"}${(Q)${:-"\~"}}${:-"%b%k%F{004}"}/${:-"%F{004}"}pipedream${:-"%b%k%F{004}"}/${${${_p9k__d:#-*}:+packages}:-${:-"%F{004}"}pa${:-"%b%k%F{004}"}${$((_p9k__d+=6))+}}/${:-"%F{004}"}component_code_gen${:-"%b%k%F{004}}${_p9k__e::=${${_p9k__1ldir+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}0}}}+}${${_p9k__e:#00}:+${${_p9k_t[$_p9k__n]/<_p9k__ss>/$_p9k__ss}/<_p9k__s>/$_p9k__s}${_p9k__c}%b%k%F{004\}${${:-${_p9k__s::=%F{\}}${_p9k__ss::= }${_p9k__sss::=%F{\} }${_p9k__i::=1}${_p9k__bg::=}}+}}${(e)_p9k__vcs}%b%k$_p9k__sss%b%k%f}}}+}${(e)_p9k_t[6]}${${_p9k__h::=0}+}${${_p9k__d::=$((_p9k__m-_p9k__h))}+}${_p9k__lprompt/\%\{d\%\}*\%\{d\%\}/${_p9k__1ldir-${:-"%F{004}"}${(Q)${:-"\~"}}${:-"%b%k%F{004}"}/${:-"%F{004}"}pipedream${:-"%b%k%F{004}"}/${${${_p9k__d:#-*}:+packages}:-${:-"%F{004}"}pa${:-"%b%k%F{004}"}${$((_p9k__d+=6))+}}/${:-"%F{004}"}component_code_gen${:-"%b%k%F{004}"}}}${${_p9k__m::=$((_p9k__d+_p9k__h))}+}${${_p9k__g+
}:-${:-"%F{002}"}${${${_p9k__m:#-*}:+${${_p9k__1g+${(pl.$((_p9k__m+1)).. .)}}:-${(pl.$((_p9k__m+1))..─.)}}$_p9k__rprompt${_p9k_t[$((1+!_p9k__ind))]}}:-
}%b%k%f}}${_p9k__2-${_p9k__2l_frame-"%2F╰─"}${_p9k__2l-${${:-${_p9k__bg::=NONE}${_p9k__i::=0}${_p9k__sss::=%f }}+}${${:-"${${:-$_p9k__keymap.$_p9k__zle_state}:#(vicmd.*|vivis.*|vivli.*|*.*overwrite*)}"}:+${${:-${P9}}${_p9k__e::=${${_p9k__2lprompt_char+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}0}}}+}${${_p9k__e:#00}:+${${_p9k_t[$_p9k__n]/<_p9k__ss>/$_p9k__ss}/<_p9k__s>/$_p9k__s}${_p9k__c}%b%k%F{002\}${${:-${_p9k__s::=%F{\}}${_p9k__ss::= }${_p9k__sss::=%F{\}}${_p9k__i::=3}${_p9k__bg::=}}+}}}${${:-"${${:-$_p9k__keymap.$_p9k__zle_state}:#(vicmd.*|vivis.*|vivli.*|*.*insert*)}"}:+${${:-${P9K_CONTENT::=▶}${_p9k__n::=}${${${_p9k__bg:-0}:#NONE}:-${}}${_p9k__e::=${${_p9k__2lprompt_char+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}0}}}+}${${_p9k__e:#00}:+${${_p9k_t[$_p9k__n]/<_p9k__ss>/$_p9k__ss}/<_p9k__s>/$_p9k__s}${_p9k__c}%b%k%F{002\}${${:-${_p9k__s::=%F{\}}${_p9k__ss::= }${_p9k__sss::=%F{\}}${_p9k__i::=3}${_p9k__bg::=}}+}}}${${:-"${(M)${:-$_p9k__keymap$_p9k__region_active}:#vicmd0}"}:+${${:-${P9K_CONTENT::=❮}${_p9k__n::=}${${${_p9k__bg:-0}:#NONE}:-${_p9k__n::=52}}${_p9k__n:}}${_p9k__e::=${${_p9k__2lprompt_char+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}0}}}+}${${_p9k__e:#00}:+${${_p9k_t[$_p9k__n]/<_p9k__ss>/$_p9k__ss}/<_p9k__s>/$_p9k__s}${_p9k__c}%b%k%F{002\}${${:-${_p9k__s::=%F{\}}${_p9k__ss::= }${_p9k__sss::=%F{\}}${_p9k__i::=3}${_p9k__bg::=}}+}}}${${:-"${(M)${:-$_p9k__keymap$_p9k__region_active}:#(vicmd1|vivis?|vivli?)}"}:+${${:-${P9K_CONTENT::=Ⅴ}${_p9k__n::=}${${${_p9k__bg:-0}:#NONE}:-${_p9k__n:}}${_p9k__e::=${${_p9k__2lprompt_char+00}:-${${(%):-$_p9k__c%1(l.1.0)}[-1]}0}}}+}${${_p9k__e:#00}:+${${_p9k_t[$_p9k__n]/<_p9k__ss>/$_p9k__ss}/<_p9k__s>/$_p9k__s}${_p9k__c}%b%k%F{002\}${${:-${_p9k__s::=%F{\}}${_p9k__ss::= }${_p9k__sss::=%F{\}}${_p9k__i::=3}${_p9k__bg::=}}+}}}%b%k$_p9k__sss%b%k%f${:-" %b%k%f"}}}${${COLUMNS::=$_p9k__clm}+}
```

Reply:
Interesting, didn't have this problem but good to know